A54-restrict-control-plane-status-codes.md

```diff
@@ -0,0 +1,136 @@
+A54: Restrict Control-Plane Status Codes
+----
+* Author(s): [Eric Anderson](https://github.com/ejona86), [Mark
+  Roth](https://github.com/markdroth), [Doug Fawley](https://github.com/dfawley)
+* Approver: [Doug Fawley](https://github.com/dfawley)
+* Status: In Review
+* Implemented in: <language, ...>
+* Last updated: 2022-07-22
+* Discussion at: https://groups.google.com/g/grpc-io/c/Yp0X2IHiVlA/m/7QyU4mYKDgAJ
+
+## Abstract
+
+The gRPC control plane has no need to use many of the status codes yet
+communicates with external services whose status codes may accidentally be
+copied to the data plane. This gRFC restricts the allowed status codes the
+control plane can use when failing an RPC, as a safeguard against bugs.
+
+## Background
+
+gRPC has a control plane consisting mainly of name resolvers and load balancers.
+Separately it has a data plane that processes individual RPCs. Error information
+from the control plane needs to pass to the data plane when errors are
+occurring, but the gRPC data plane is restricted from generating [status codes
+reserved for application use][statuscodes.md]: `INVALID_ARGUMENT`, `NOT_FOUND`,
+`ALREADY_EXISTS`, `FAILED_PRECONDITION`, `ABORTED`, `OUT_OF_RANGE`, and
+`DATA_LOSS`. Those status codes are reserved for each service.
+
+Earlier in gRPC's life, there was relatively low risk of using a reserved status
+code in the data plane. There was a relatively fixed amount of control plane
+code, and much of it wasn't accessing gRPC-based services (e.g., DNS,
+pick_first, round_robin). At that point in time, a simple audit could reveal
+broken status code usages.
+
+Now, however, it is more common to implement custom name resolvers and load
+balancers and for those components to use gRPC themselves. This happened first
+with gRPC-LB, but also happens with xDS, Route Lookup Service (RLS), and others.
```

Comment:
We've never publically documented RLS, so it's probably best not to mention it explicitly here.  I think it's fine to say something like "also happens with xDS, and will happen with other control planes in the future".